### PR TITLE
IN-453 

### PR DESCRIPTION
**Pull request by jmackown**
## Purpose
Two things, because I accidentally did them at the same time (sorry)
1. removed `generate_select_string` from the mapping class, so it can be used standalone
2. Switch to using a simpler format for the mapping. Gone from:
```
{
	"simple_mapping": {
		"orderdate": {
			"casrec_table": "ORDER",
			"casrec_column_name": "Made Date",
			"alias": "Made Date",
		},
		"caserecnumber": {
			"casrec_table": "ORDER",
			"casrec_column_name": "Case",
			"alias": "Case",
		}
	},
	"transformations": {
		"unique_number": [{
			"original_columns": ["unknown"],
			"aggregate_col": "uid"
		}]
	},
	"required_columns": {
		"type": {
			"data_type": "str",
			"default_value": "order"
		}
	},
}
```

to a much simpler:
```
{
        "caserecnumber": {
            "casrec_table": "PAT",
            "casrec_column_name": "Case",
            "alias": "Case",
        },
        "casrec_id": {
            "casrec_table": "pat",
            "casrec_column_name": "rct",
            "alias": "rct",
        },
        "dob": {
            "casrec_table": "pat",
            "casrec_column_name": "DOB",
            "transformation": "date_format_standard",
        },
        "uid": {
            "casrec_table": "",
            "casrec_column_name": "",
            "new_column": "uid",
            "transformation": "unique_number",
        },
}
```

### CODE REVIEW

File: etl2/app/mapping/mapping.py [hunk 1]
```diff
@@ -42,143 +26,54 @@ def _apply_column_alias(self, df):
...
-        )
-        mapping_definitions["required_columns"] = self.required_columns(
-            full_mapping=mapping_df
+        interesting_columns = [
+            "casrec_table",
+            "casrec_column_name",
+            "alias",
+            "requires_transformation",
+            "default_value",
+            # 'is_pk',
```
Comment (jamesrwarren) [inline etl2/app/mapping/mapping.py:45]: these supposed to be here?

File: etl2/app/mapping/mapping.py [hunk 2]
```diff
@@ -1,35 +1,19 @@
...
-        #     config.row_limit
-        #     if config.row_limit is not None
-        #     else None
-        # )
+        self.sheet_name = sheet_name
 
     def _read_mapping_doc(self):
         sheet = pd.read_excel(io=self.excel_doc, sheet_name=self.sheet_name)
 
+        # print(f"\n{sheet.to_markdown()}")
```
Comment (jamesrwarren) [inline etl2/app/mapping/mapping.py:15]: ditch comment

File: etl2/app/mapping/mapping.py (hunk 1 above)
Comment (jmackown) [reply etl2/app/mapping/mapping.py:45]: They are leftover from me pruning out the next bit. They will appear again...

File: etl2/app/mapping/mapping.py (hunk 2 above)
Comment (jmackown) [reply etl2/app/mapping/mapping.py:15]: removed

**Review by jamesrwarren — APPROVED** [review]

**Pull request merged**